core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java

```diff
@@ -1088,20 +1099,33 @@ private List<TemplateParameterInfo> gatherTemplateParamInfo(List<MethodParameter
         return templateParams;
     }
 
-    private List<AnnotationInstance> effectiveParamAnnotations(MethodParameterInfo param) {
-        return param.annotations().stream().filter(ai -> {
-            String name = ai.name().toString();
-            if (name.startsWith("kotlin") || name.startsWith("jakarta.validation.constraints")) {
-                return false;
-            }
-            if (name.endsWith("NotNull")) {
-                return false;
-            }
-            if (name.startsWith("io.opentelemetry")) {
-                return false;
+    private boolean isParameterAllowedAsTemplateVariable(
+            MethodParameterInfo param, Collection<String> allowedPrefixes) {
+
+        Collection<MethodParameterAsTemplateVariableAllowance> allowances = param.annotations().stream().map(anno -> {
+            String name = anno.name().toString();
+
+            if (allowedPrefixes.stream().anyMatch(prefix -> name.startsWith(prefix))) {
+                // Any annotations matching allowed prefix are allowed
+                return FORCE_ALLOW;
+            } else {
+                // Any annotations matching predicates below are ignored
+                if (name.startsWith("kotlin") || name.startsWith("jakarta.validation.constraints")) {
+                    return IGNORE;
+                }
+                if (name.endsWith("NotNull")) {
+                    return IGNORE;
+                }
+                if (name.startsWith("io.opentelemetry")) {
+                    return IGNORE;
```

Comment:
Wouldn't it make sense to have a new Build Item that configures these `IGNORE` values in a similar way as `MethodParameterAllowedAnnotationsBuildItem` configures `FORCE_ALLOW`?

Reply:
Not sure, it might probably be useful later. Probably beyond the scope of this pr ?

Reply:
Yeah, you are probably right